app/api/helpers/permissions.py

```diff
@@ -157,10 +158,18 @@ def is_coorganizer(f):
     def decorated_function(*args, **kwargs):
         user = current_user
 
-        if user.is_staff:
-            return f(*args, **kwargs)
-        if 'event_id' in kwargs and user.has_event_access(kwargs['event_id']):
+        if 'event_identifier' in kwargs:
+            if not kwargs['event_identifier'].isdigit():
+                event = db.session.query(Event).filter_by(identifier=kwargs['event_identifier']).first()
```

Comment:
As said previously. This should not be done here. This is a violation of SRP

Reply:
@iamareebjamal I am not returning `event_id` here, this is just for functionality of this function.

Reply:
converting from event_identifier to event_id should be done in a separate decorator so things remain DRY and serve a single responsibility.

Current code requires 2 DB hits to change identifier to ID

Reply:
I also thought that would be the best way to go. But got stuck into some really ugly error logs. Will give it one more try and ask here if stuck.